### ISSUE TITLE
Timer stops ticking after ~30s when app is backgrounded

### ISSUE THREAD

**Issue by dgandle** [dgandle/pomoRN#3]
On iOS, the timer stops after about 30 seconds or so when the app is backgrounded. This is due to the way React Native uses iOS's dispatch queues (more info [here](https://github.com/ocetnik/react-native-background-timer/issues/222#issuecomment-628782753)) and fixing it will likely require accessing GCD with native modules. 

Assuming that GH issues linked above is correct, the background timer is being scheduled on the iOS main thread, which stop executing after ~30s of the app being backgrounded (seems like this number could change depending on low battery mode, etc). My proposed solution is to create a native module for iOS that would start a background timer (instead of using react-native-background-timer) and schedule it on a background thread.

**Comment by jothikannan** [comment]
Yes I am also having the same issue when the app state goes to background,   `BackgroundTimer.runBackgroundTimer` call never calling
